extensions/ql-vscode/src/view/variant-analysis/__tests__/VariantAnalysis.spec.tsx

```diff
@@ -46,4 +65,33 @@ describe(VariantAnalysis.name, () => {
       ),
     ).toBeInTheDocument();
   });
+
+  it("renders results view with correct filter and sort state", async () => {
+    const variantAnalysis = createMockVariantAnalysis({});
+    render({ variantAnalysis });
+
+    // Without this await, `getByDisplayValue` could not find any selected dropdown option.
+    await new Promise((resolve) => setTimeout(resolve, 0));
+
+    expect(screen.getByDisplayValue("With results")).toBeInTheDocument();
+    expect(screen.getByDisplayValue("Number of results")).toBeInTheDocument();
```

Comment:
For some reason, `await new Promise((resolve) => setTimeout(resolve, 0));` was necessary to get these "default" expectations to pass 🤔 

Not sure why, but it works 😅 If anyone knows more about this, let us know!

Reply:
I'm not sure why this is happening (it could be that some of the VS Code UI toolkit shadow DOM is only rendered on the next tick or something like that), but a slightly more reliable way to solve this is by waiting for the element to start existing:

```ts
await waitFor(() => screen.getByText("With results"));
```

This way we don't depend on the exact tick and it will still fail if it doesn't find the element after some time.

Reply:
This is perfect! Thanks ✨